Fix bug in `create` where command line arguments were truncated

azeey opened the pull request:
# 🦟 Bug fix

## Summary

Arguments were being truncated because the size argument of `sprintf` was `sizeof(filtered_argv)`, which is the size of a pointer (e.g 8).

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

Commit:
Fix bug in `create` where command line arguments were truncated

Signed-off-by: Addisu Z. Taddese <addisu@openrobotics.org>

ahcorde reviewed (changes requested):
ups 

```bash
/github/home/ws/src/ros_gz/ros_gz_sim/src/create.cpp:63:  Almost always, snprintf is better than strcpy  [runtime/printf] [4]
```

Commit:
Add test

Signed-off-by: Addisu Z. Taddese <addisu@openrobotics.org>

Commit:
Use sprintf instead of strcpy

Signed-off-by: Addisu Z. Taddese <addisu@openrobotics.org>

azeey commented:
> ups
> 
> ```shell
> /github/home/ws/src/ros_gz/ros_gz_sim/src/create.cpp:63:  Almost always, snprintf is better than strcpy  [runtime/printf] [4]
> ```

Okay, changed it back to `snprintf`. Also added a test. I haven't used launch_testing before, so let me know if I'm not using it correctly.

ahcorde reviewed (changes requested):
linters are failing
```bash
2023-12-20T21:27:06.0815611Z The following tests FAILED:
2023-12-20T21:27:06.0817056Z 	  5 - cpplint (Failed)
2023-12-20T21:27:06.0819323Z 	  6 - flake8 (Failed)
2023-12-20T21:27:06.0821402Z 	  9 - uncrustify (Failed)
```

Commit:
Fix linters

Signed-off-by: Addisu Z. Taddese <addisu@openrobotics.org>
